hyper/http11/connection.py

```diff
@@ -75,6 +79,19 @@ def __init__(self, host, port=None, secure=None, ssl_context=None,
         self.ssl_context = ssl_context
         self._sock = None
 
+        # Setup proxy details if applicable.
+        if proxy_host:
+            if proxy_port is None:
+                try:
+                    self.proxy_host, self.proxy_port = proxy_host.split(':')
```

Comment:
Can we pull this line out of the `try...except`? It isn't the one raising the `ValueError`, and tightly scoped `try` blocks are good.